src/components/Animation/Controllers.tsx

```diff
@@ -0,0 +1,146 @@
+
+			if (event.currentTarget) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => {
+					child.classList.remove('active')
+				})
+
+				if (target.className === 'count' || target.className === 'counter') {
+					target.parentElement?.classList.add('active')
+					setCurrentRange(childs.findIndex(item => item.className.includes('active')))
+					return
+				}
+
+				target.classList.add('active')
+				setCurrentRange(childs.findIndex(item => item.className.includes('active')))
+			}
+		}
+
+		const handleOnMouseOver = (event: React.MouseEvent<HTMLElement>) => {
+			const target = event.target as Element
+
+			if (event.currentTarget) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => child.classList.remove('active'))
+
+				if (target.className === 'count' || target.className === 'counter') {
+					target.parentElement?.classList.add('active')
+					return
+				}
+
+				target.classList.add('active')
+			}
+		}
+
+		const handleOnMouseOut = (event: React.MouseEvent<HTMLElement>) => {
+			if (event.currentTarget && items) {
+				const childs = [...event.currentTarget.children]
+				childs.forEach((child) => child.classList.remove('active'))
+				items.children[currentRange].classList.add('active')
+			}
+		}
+
+		if (items) {
+			items.addEventListener('click', handleOnClick)
+
+			items.addEventListener('mouseover', handleOnMouseOver)
+
+			items.addEventListener('mouseout', handleOnMouseOut)
+		}
+
+		return () => {
+			items.removeEventListener('click', handleOnClick)
+
+			items.removeEventListener('mouseover', handleOnMouseOver)
+
+			items.removeEventListener('mouseout', handleOnMouseOut)
+		}
+	}, [currentRange, classId])
+
+	return (
+		<div
+			key={`${+keyToggle.current}-ranges`}
+			className={'animation-handlers'}
+		>
+			<div className='animation-handlers__item'>
+				<ParticlesCounter
+					key={`${+keyToggle.current}-particles`}
+					initialValue={properties.particleCount}
+					onChange={changeParticlesCount}
+				/>
+			</div>
+
+			{classId !== SpiralFlyingLines.name && (
+				<div className='animation-handlers__item'>
+					<LineLengthCounter
+						key={`${+keyToggle.current}-length`}
+						initialValue={properties.lineLength}
+						onChange={changeLineLength}
+					/>
+				</div>
+			)}
+
+			{classId !== Spiral.name && classId !== SpiralFlyingLines.name && (
+				<div className='animation-handlers__item'>
+					<VelocityCounter
+						key={`${+keyToggle.current}-velocity`}
+						initialValue={properties.particleMaxVelocity}
+						onChange={changeVelocity}
+					/>
+				</div>
+			)}
+
+			{classId === SpiralFlyingLines.name &&
+        properties.radius &&
+        !properties.isPulsatile && (
+				<div className='animation-handlers__item'>
+					<RadiusCounter
+						key={`${+keyToggle.current}-radius`}
+						initialValue={properties.radius || 0}
+						onChange={changeRadius}
+					/>
+				</div>
+			)}
+
+		</div>
```

Comment:
**Please**, check [here](https://github.com/kstroevsky/bindfly.js/blob/main/src/components/Animation/index.tsx#L61) how I mapped this components previously.